examples/embeddable_examples/public/plugin.ts

```diff
@@ -100,9 +109,30 @@ export class EmbeddableExamplesPlugin
     );
   }
 
-  public start(core: CoreStart, deps: EmbeddableExamplesStartDependencies) {
+  public start(
+    core: CoreStart,
+    deps: EmbeddableExamplesStartDependencies
+  ): EmbeddableExamplesStart {
     return {
       createSampleData: () => createSampleData(core.savedObjects.client),
+      helloWorldEmbeddableFactory: deps.embeddable.getEmbeddableFactory(
```

Comment:
Initially I was looking into something like this to avoid typecasting: 

```tsx

this.factory = deps.embeddable.registerEmbeddableFactory(
       TODO_EMBEDDABLE,
       new TodoEmbeddableFactory()
);

// ....

return {
  todoEmbeddable: this.factory
}
```

But looks like we can't do this, because we want factories to be created in start phase. (for example, because we can change defaultFactory for embeddables)

Reply:
One way to do this is to have `registerEmbeddableFactory` return a function that returns the factory after the enhnacement is registered.

```
  registerEmbeddableFactory: <I extends EmbeddableInput, O extends EmbeddableOutput>(
    id: string,
    factory: EmbeddableFactoryDefinition<I, O>
  ) => () => Promise<EmbeddableFactory<I, O>>;
```

https://github.com/elastic/kibana/pull/62998/files#diff-af21272406d0e6b0842f57e4282c326cR98

^^ That example doesn't use a promise, but throws an exception if accessed at the incorrect time.

Reply:
If that is too much work though, I think this is still better. At least the registrator is doing the casting, not the consumer.

Reply:
@stacey-gammon, I went for this, I like it better now. 
Only thing, I went with not promise version and throwing error if accessed during setup, to not make downstream code async